<?xml version="1.0"?>
<launch>
    <rosparam command="load" file="$(find kuam_aruco_tracking)/config/param.yaml" />

    <!-- Select only one data type -->
    <arg name="simulation" default="false" />

    <!-- Aruco lib param -->
    <arg name="detector_params_path" default="$(find kuam_aruco_tracking)/config/detector_params.yml" />
    <arg name="calib_path" default="$(find kuam_aruco_tracking)/config/calib_sim.txt" if="$(arg simulation)"/>
    <arg name="calib_path" default="$(find kuam_aruco_tracking)/config/calib.txt" unless="$(arg simulation)"/>

    <!-- rviz -->
    <arg name="rviz" default="false" />

    <!-- nodes -->
    <group ns="aruco_tracking">
        <node pkg="image_proc" type="image_proc" name="image_proc" />
    </group>


    <node pkg="kuam_aruco_tracking" type="aruco_tracking_node" name="aruco_tracking" cwd="node" output="screen">
        <param name="calib_path" value="$(arg calib_path)" />
        <param name="detector_params_path" value="$(arg detector_params_path)" />
        <param name="simulation" value="$(arg simulation)" />
    </node>
    <node pkg="kuam_aruco_tracking" type="aruco_tf_node" name="aruco_tf" output="screen" />
    <node pkg="kuam_aruco_tracking" type="aruco_visual_node" name="aruco_visual" />
    <group unless="$(arg simulation)">
        <include file="$(find kuam_usb_cam)/launch/usb_cam.launch" />
    </group>

    <group if="$(arg rviz)">
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(find kuam_aruco_tracking)/rviz/aruco_marker.rviz" required="true" />
    </group>
</launch>
